### PR TITLE
Add support for healthcheck in compose to swarm

### LINKED ISSUE #28520: compose on swarm: update parallelism needs better default
If I don't specify an `update_config` in my compose file, I get the defaults, which is great.

If I specify something like this:

```
     deploy:
       update_config:
         failure_action: pause
```

...then I get an `UpdateConfig` in the service, but the other fields are initialized to 0. For `Parallelism`, this is inconsistent with the CLI, where it defaults to 1.

I believe the other fields in `UpdateConfig` are handled correctly. `RestartConfig` and other structs also look okay.

cc @dnephin @vdemeester @aluzzardi

### PR DESCRIPTION
**Pull request by vdemeester**
Fixes #28520

Adds support for `healtcheck` in the composefile v3 format, used it `docker stack deploy --compose-file` 🐮.

```yaml
version: "3"

services:
  foo:
    image: busybox
    command: top
    healthcheck:
      test: cat /etc/password
      interval: 10s
      timeout: 2s
      retries: 3
```

/cc @dnephin @aanand @icecrime 

This bumps [compose-file](https://github.com/aanand/compose-file) to a70cb6ea5c403c383b5117dcd1dea78880a47824.

🐸

### CODE REVIEW

File: cli/command/stack/deploy.go [hunk 1]
```diff
@@ -520,6 +528,45 @@ func convertService(
...
+	}
+	var (
+		err               error
+		test              []string
+		timeout, interval time.Duration
+		retries           int
+	)
+	// FIXME(vdemeester) decide what mean NONE for healthcheck
+	if len(healthcheck.Command) == 0 || healthcheck.Command[0] == "none" {
+		test = []string{"NONE"}
```
Comment (cpuguy83) [inline cli/command/stack/deploy.go]: I'd not use a magic value.
Something like `DisableHealthCheck: true` seems good to me.
Then someone can just add or remove this to the existing config to enable/disable.

Comment (vdemeester) [inline cli/command/stack/deploy.go]: /cc @aanand @dnephin

Comment (dnephin) [inline cli/command/stack/deploy.go]: hmm, ya I guess we could do:

```yaml
healthcheck:
  disable: true
```

And then error if any other fields are set at the same time as `disable: true` .

Comment (vdemeester) [inline cli/command/stack/deploy.go]: sounds good to me 👼

Comment (cpuguy83) [inline cli/command/stack/deploy.go]: I don't think we'd want to error, would we? Then people can't turn it on/off quickly.

Comment (dnephin) [inline cli/command/stack/deploy.go]: Yes, this needs to be an error:

```yaml
healthcheck:
  disable: true
  command: foo
```

That config doesn't make any sense. It doesn't make any more time to comment out the `command` line.

Comment (vdemeester) [inline cli/command/stack/deploy.go]: I'll do a PR in compose-file for that then (/cc @aanand for the `docker/compose` part)

**Comment by vdemeester** [comment]
@aanand @dnephin updated with latest changes (`commant` -> `test`, `disable`, …)

File: cli/command/stack/deploy.go [hunk 2]
```diff
@@ -531,6 +539,47 @@ func convertExtraHosts(extraHosts map[string]string) []string {
...
+		if err != nil {
+			return nil, err
+		}
+	}
+	if healthcheck.Interval != "" {
+		interval, err = time.ParseDuration(healthcheck.Interval)
+		if err != nil {
+			return nil, err
+		}
+	}
```
Comment (dnephin) [inline cli/command/stack/deploy.go]: I guess these could be `time.Duration` in the compose-file/types, but we can fix that alter.

**Review by dnephin — APPROVED** [review]
LGTM

File: cli/command/stack/deploy.go (hunk 2 above)
Comment (vdemeester) [inline cli/command/stack/deploy.go]: Ah good point :stuck_out_tongue:

**Comment by vieux** [comment]
LGTM

**Pull request merged**